src/App.jsx

```diff
@@ -2,59 +2,39 @@ import React, { Component } from 'react';
 import './style.css';
 import 'materialize-css';
 import 'materialize-css/dist/css/materialize.min.css';
-import { loadMovies } from './actions/actionCreator';
+import { clearStore, loadMovies, selectMovie } from './actions/actionCreator';
 import { connect } from 'react-redux';
 import SearchPage from './components/SearchPage';
 import MoviePage from './components/MoviePage';
+import NotFoundPage from "./components/NotFoundPage";
+import { BrowserRouter, Route, Switch } from "react-router-dom";
+import PropTypes from 'prop-types';
 
 class App extends Component {
 
-  componentDidMount() {
-    this.props.loadMovies();
-  };
-
   render() {
     return (
       <React.Fragment>
-        <SearchPage
-          movies={this.props.movies}
-          searchby={this.props.searchby}
-          sortby={this.props.sortby} />
-        {/* <MoviePage movies={this.state.data}/> */}
+        <BrowserRouter>
+          <div>
+            <Switch>
+              <Route exact path='/' component={SearchPage} />
+              <Route path='/search/:query' component={SearchPage} />
+              <Route path='/film/:id' component={MoviePage} />
+              <Route path='*' component={NotFoundPage} />
+            </Switch>
+          </div>
+        </BrowserRouter>
       </React.Fragment>
     );
   }
 }
 
-const mapDispatchToProps = { loadMovies };
-
-const selectSortedMovies = (state) => {
-  const sortby = state.sortby;
-  const movies = state.movies.data;
-
-  const sortbyDateOrRating = {
-    DATE: "release_date",
-    RATING: "vote_average"
-  }[sortby];
-
-  const dataForSort = (data) => {
-    if (sortbyDateOrRating === "release_date") {
-      return data.split("-").join("");
-    }
-    return data;
-  };
-
-  return movies.sort((a, b) => {
-    return dataForSort(b[sortbyDateOrRating]) - dataForSort(a[sortbyDateOrRating]);
-  });
+App.propTypes = {
+  loadMovies: PropTypes.func,
+  selectMovie: PropTypes.func,
```

Comment:
Are these props used in this component? Please check and update.

Reply:
Updated.